src/hooks/useUsersGuildPins.ts

```diff
@@ -91,20 +181,40 @@ const fetchGuildPins = async ([_, addresses]: [string, User["addresses"]]) => {
     )
   )
 
-  return responseArray.flat()
+  const { allUsersPins, allErrors } = responseArray.reduce(
+    (acc, response) => {
+      if (response.tokenMetadata) {
+        acc.allUsersPins.push(...response.tokenMetadata)
+      }
+      if (response?.errors.length > 0) {
+        acc.allErrors.push(...response.errors)
+      }
+      return acc
+    },
+    { allUsersPins: [] as TokenWithMetadata[], allErrors: [] as Error[] }
+  )
+
+  return { usersPins: allUsersPins, errors: allErrors }
 }
 
-const useUsersGuildPins = (disabled = false) => {
+const useUsersGuildPins = (disabled = false, includeTestnets = false) => {
   const { isWeb3Connected } = useWeb3ConnectionManager()
   const { addresses } = useUser()
 
   const evmAddresses = addresses?.filter((address) => address.walletType === "EVM")
 
   const shouldFetch = Boolean(!disabled && isWeb3Connected && evmAddresses?.length)
 
-  return useSWRImmutable<
-    ({ chainId: number; tokenId: number } & GuildPinMetadata)[]
-  >(shouldFetch ? ["guildPins", evmAddresses] : null, fetchGuildPins)
+  const swrData = useSWRImmutable(
+    shouldFetch ? ["guildPins", addresses, includeTestnets] : null,
+    fetchGuildPins
+  )
+
+  return {
+    ...swrData,
+    data: swrData.data?.usersPins,
+    pinFetchErrors: swrData.data?.errors,
```

Comment:
I think we could just leave this prop out and instead return a boolean in the `error` prop (`swrData.data?.errors?.length > 0`), since we won't use the actual errors anywhere